apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Query/MessageArchiveQuery.cs

```diff
@@ -56,6 +62,93 @@ public async Task<IEnumerable<ArchivedMessageResult>> GetArchivedMessagesAsync(
                     : businessReasons?.Select(x => x.ToString()).ToArray(),
             };
 
-        return await client.ArchivedMessageSearchAsync("1.0", search);
+        var (field, direction) = order switch
+        {
+            { MessageId: not null } => (FieldToSortBy.MessageId, order.MessageId),
+            { DocumentType: not null } => (FieldToSortBy.DocumentType, order.DocumentType),
+            { Sender: not null } => (FieldToSortBy.SenderNumber, order.Sender),
+            { Receiver: not null } => (FieldToSortBy.ReceiverNumber, order.Receiver),
+            { CreatedAt: not null } => (FieldToSortBy.CreatedAt, order.CreatedAt),
+            _ => (FieldToSortBy.CreatedAt, SortDirection.Desc),
+        };
```

Comment:
if you update ArchivedMessageSortInput as commented. Then this should not be needed any more